### ISSUE TITLE
Using transform-origin cross-platform

### ISSUE THREAD

**Issue by foodelix-admin** [tamagui/tamagui#1415]
### Current Behavior

T in consideration a component like the following if you want to animate the height of a component

```
      <Stack
        animation={'quick'}
        width={200}
        scaleY={isCollapsed ? 0 : 1}
        bg="red"
      />
```
it woks very well, and also cross-platform (AMAZING!), but it will animate starting from center. 

Let say now taht I want to use that Stack to create a collapsible panel, I may want to animate it using `top-center` as the transform origin, but unfortunately the only way I found to do this is by adding `style={{ transformOrigin: 'center top' }}`, which breaks cross-platform compatibility.

The only way I found to do it cross platform, is to directly use a RN's `Animated.View`, and use Animated values to control the height, but by using `Animated.View` I lose all tamagui style props benefits.

Is there a better way to do it ? Or given this requirements it is the only way to accomplish this?

### Expected Behavior

There should be an easier way to animate width and height cross platform

### Tamagui Version

```markdown
latest as of 12/07/2023
```

### Reproduction

```markdown
There is nothing to reproduce here, it just a "conceptual" issue
```

### System Info

```markdown
System:
    OS: macOS 13.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 321.50 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 3.5.0 - ~/.nvm/versions/node/v16.17.0/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
  Browsers:
    Brave Browser: 112.1.50.119
    Chrome: 114.0.5735.198
    Firefox: 112.0.2
    Firefox Developer Edition: 113.0
    Safari: 16.5
```

**Comment by natew** [comment]
does animated.view accept transform origin?

**Comment by foodelix-admin** [comment]
No, it doesn't, but it does animate height if passed as animated value, on the other hand tamagui doesn't animate height if component has animation prop, you can see what I mean in [this](https://github.com/foodelix-admin/tamagui-animation-bug ) repo where I reproduced the issue on the home screen

**Comment by natew** [comment]
React Native has a PR open for this, once that is merged we can support it.
